w_common/example/disposable/example.dart

```diff
@@ -32,28 +32,31 @@ void main() {
 
   Disposable.enableDebugMode();
 
-  ButtonElement createButton = querySelector('#create-button');
-  ButtonElement disposeButton = querySelector('#dispose-button');
-  InputElement childCountField = querySelector('#child-count-field');
-  InputElement treeDepthField = querySelector('#tree-depth-field');
+  ButtonElement createButton = querySelector('#create-button') as ButtonElement;
+  ButtonElement disposeButton =
+      querySelector('#dispose-button') as ButtonElement;
+  InputElement childCountField =
+      querySelector('#child-count-field') as InputElement;
+  InputElement treeDepthField =
+      querySelector('#tree-depth-field') as InputElement;
 
-  TreeNode treeRoot;
+  TreeNode? treeRoot;
 
   createButton.onClick.listen((_) {
     if (treeRoot != null) {
       window.alert('Dispose before creating a new tree');
     }
 
-    int childCount = int.parse(childCountField.value);
-    int treeDepth = int.parse(treeDepthField.value);
+    int childCount = int.parse(childCountField!.value!);
+    int treeDepth = int.parse(treeDepthField!.value!);
 
     treeRoot = TreeNode(treeDepth, childCount);
-    print('Disposable tree size: ${treeRoot.disposalTreeSize}');
+    print('Disposable tree size: ${treeRoot!.disposalTreeSize}');
   });
 
   disposeButton.onClick.listen((_) {
-    treeRoot?.dispose()?.then((_) {
-      print('Disposable tree size: ${treeRoot.disposalTreeSize}');
+    treeRoot?.dispose().then((_) {
```

Comment:
Why did the ? after dispose go away? Is analyzer not complaining about this?

Reply:
https://dart.dev/null-safety/understanding-null-safety#smarter-null-aware-methods